apps/passport-client/src/dispatch.ts

```diff
@@ -576,17 +576,48 @@ async function makeUploadId(
  *   to e2ee, then uploads then to e2ee.
  */
 async function sync(state: AppState, update: ZuUpdate) {
+  // This re-entrancy protection ensures only one event-sequence is
+  // inside of doSync() at any time.  When doSync() completes, it will
+  // run again if any state changes were made.
+  if (!syncInProgress) {
+    try {
+      syncInProgress = true;
+      const stateChanges = await doSync(state, update);
+      if (stateChanges) {
+        update(stateChanges);
+      }
+    } finally {
+      syncInProgress = false;
+    }
+  }
+}
+
+// TODO(artwyman): Is there a better way to do this than a global
+// variable?
+let syncInProgress = false;
```

Comment:
I'm not sure that it's a huge problem that we need to avoid

Reply:
What problem, global variables?  They're an antipattern in a lot of my experience, but maybe they're more normal in a web-app context?